File: VirtualWorld-Python/zwierzeta/Wilk.py
from Zwierze import Zwierze

class Wilk(Zwierze):
    
    def __init__(self):
        pass

    def __init__(self, swiat, poz, wiek, inicj, moc):

        super(Zwierze, self).__init__()

        super().changeSwiat(swiat)
        super().changePolozenie(poz[0], poz[1])
        super().changeWiek(wiek)
        super().changeInicjatywa(inicj)

        s = "W"
        super().getSwiat().edytujPlansze(poz[0], poz[1], s)

    def __init__(self, swiat, poz):

        super(Zwierze, self).__init__()

        super().changeSwiat(swiat)
        super().changePolozenie(poz[0], poz[1])
        super().changeInicjatywa(5)
        super().zmienSile(9)
        super().changeWiek(0)

        s = "W"
        super().getSwiat().edytujPlansze(poz[0], poz[1], s)

    #--------------------------------------------

    def rysowanie(self):
        print('W')

    def getZnak(self):
        return 'W'

File: VirtualWorld-Java/src/Wilk.java

import java.util.ArrayList;


public class Wilk extends Zwierze{
	
	public Wilk() {
		
	}
	
	public Wilk(Swiat swiat, ArrayList<Integer> poz, int wiek, int inicj, int moc) {
		changeSwiat(swiat);
		changePolozenie(poz.get(0), poz.get(1));
		changeWiek(wiek);
		changeInicjatywa(inicj);
		
		String s = ""+'W';
		getSwiat().edytujPlansze(poz.get(0), poz.get(1), s);
	}
	
	public Wilk(Swiat swiat, ArrayList<Integer> poz) {
		changeSwiat(swiat);
		changePolozenie(poz.get(0), poz.get(1));
		changeInicjatywa(5);
		zmienSile(9);
		changeWiek(0);
		
		String s = ""+'W';
		getSwiat().edytujPlansze(poz.get(0), poz.get(1), s);
	}
	
	@Override
	public void rysowanie() {
		System.out.println('W');
	}
	
	@Override
	public char getZnak() {
		return 'W';
	}
	
}
